adam_core/coordinates/cometary.py

```diff
@@ -407,31 +200,39 @@ def to_cartesian(self) -> CartesianCoordinates:
             )
             raise ValueError(err)
 
+        # Extract gravitational parameter from origin
+        mu = self.origin.mu
+
         coords_cartesian = cometary_to_cartesian(
-            self.values.filled(),
-            t0=self.times.tdb.mjd,
-            mu=self.mu,
+            self.values,
+            t0=self.times.to_astropy().tdb.mjd,
+            mu=mu,
             max_iter=100,
             tol=1e-15,
         )
         coords_cartesian = np.array(coords_cartesian)
 
-        if self.covariances is not None:
+        cometary_covariances = self.covariances.to_matrix()
+        if not np.all(np.isnan(cometary_covariances)):
             covariances_cartesian = transform_covariances_jacobian(
                 self.values,
-                self.covariances,
+                cometary_covariances,
                 _cometary_to_cartesian,
                 in_axes=(0, 0, None, None, None),
                 out_axes=0,
-                t0=self.times.tdb.mjd,
-                mu=self.mu,
+                t0=self.times.to_astropy().tdb.mjd,
+                mu=mu,
                 max_iter=100,
                 tol=1e-15,
             )
         else:
-            covariances_cartesian = None
+            covariances_cartesian = np.zeros(
```

Comment:
```suggestion
            covariances_cartesian = np.empty(
```

Reply:
Fixed for all coordinate representations.